### ISSUE TITLE
Integrated PCB design

### ISSUE THREAD

**Issue by amcewen** [museuminabox/brain#86]
An issue to collect thoughts/ideas/wishlist items for the next revision of the custom PCB.

Combining the amp and possibly an i2s DAC with the progress PCB could reduce the amount of wiring we have to do in assembling the box, and would improve the availability (and maybe the lead time) for sourcing amp circuits.

Input options would be basically the same as we have now: a bunch of IO pins to toggle LEDs, and either stereo analogue audio or i2s audio.

Output would also be the same - 3.5mm stereo audio jack and a single speaker (I imagine we could cope with two speakers if the design required it)

I think the analogue volume control is also important.

**Comment by amcewen** [comment]
Some amp options (as the YDA138 chip our current design uses seems to be hard to source):
 * TPA3136 - no volume option, would need to add that separately; easy-to-solder package; good documentation for mono speaker operation
 * TDA7491 - volume control; stereo amp with no indication of mono circuit; reasonably easy-to-solder package
 * MAX9768 - volume control; mono amp; hard (it's 4mm x 4mm in size) to solder package

All of these options are surface mount.  If we move to that for the amp, is it worth doing the same for the LEDs?  That might make sense anyway (with a suitable right-angle light pipe) if we need to line up the LEDs with the centre-line of the potentiometer.  It would mean the LEDs didn't protrude like they do now, and the green ready LED wouldn't be larger, but that has caused confusion in deployment, so that might not be a bad thing...

**Comment by amcewen** [comment]
Having sourced another YDA138 amp, and checked the [datasheet for it](https://pdf1.alldatasheet.com/datasheet-pdf/view/205149/YAMAHA/YDA138.html) it looks like our existing supply was lying about the chip it uses.  According to datasheet, the YDA138 operates from 9-12V (and the one I bought bears that out - if I drop is supply below 9V it stops outputting any sound), and is also a bigger chip (it's possible that it's available in different packages, although the correct chip has more pins too...).  Annoyingly the chip on the existing amps we were using has been scraped off by the supplier, so i can't easily work out what it is.

The TPA3116 chip looks like it could be another candidate.  There are some boards available which use this:
 * [Mono, barrel-jack for power supply, claims it needs 8-25V so not clear it'll run off 5V](https://www.ebay.co.uk/itm/TPA3116-100W-Subwoofer-Digital-Power-Amplifier-Board-TPA3116D2-Amplifiers-ZY/123060744101?epid=17010262556&hash=item1ca6fdefa5%3Ag%3AQUQAAOSwcB5ZNQFz&_sacat=0&_nkw=tpa3116&_from=R40&rt=nc&_trksid=p2380057.m570.l1313.TR8.TRC2.A0.H0.Xtpa3116.TRS0)
 * [Stereo, screw terminals for power, claims to run from 12V-24V so not clear it'll run off 5V](https://www.ebay.co.uk/itm/TPA3116-2x100W-Dual-Channel-Audio-Speaker-Digital-Amplifier-Board-12V-24V-AT7/253511403604?hash=item3b06750854%3Ag%3A-5IAAOSwmOdas0Q5&_sacat=0&_nkw=tpa3116&_from=R40&rt=nc&_trksid=p2380057.m570.l1313.TR8.TRC2.A0.H0.Xtpa3116.TRS0)
 * [Stereo, headers for all connections, claims to be 12-24V supply, but tiny board](https://www.ebay.co.uk/itm/TPA3116D2-50W-2-DC-12-24V-Digital-Amplifier-Board-Class-Dual-Channel-Stereo-AMP/183302562860?_trkparms=aid%3D555018%26algo%3DPL.SIM%26ao%3D2%26asc%3D52945%26meid%3D7b8dae15118b4b8bb7ccd9a23d68f779%26pid%3D100005%26rk%3D3%26rkt%3D12%26sd%3D163043551899%26itm%3D183302562860&_trksid=p2047675.c100005.m1851)
 * [Stereo, screw terminals for power, 5-24V supply](https://www.ebay.co.uk/itm/TPA3116D2-Dual-50W-Channel-Stereo-Digital-Amplifier-Board-Power-Amp-5-24V-TE743/163043551899?hash=item25f6272a9b%3Ag%3AbekAAOSwIphbMXdf&_sacat=0&_nkw=tpa3116&_from=R40&rt=nc&_trksid=p2380057.m570.l1313.TR8.TRC2.A0.H0.Xtpa3116.TRS0)
 * [Mono, screw terminals, would need the volume potentiometer adding - but that would allow us to choose a squarer one - also claims only 12/24V supply](https://www.ebay.co.uk/itm/TPA3116-D2-DA-DC-12V-24V-100W-Mono-Channel-Digital-Power-Audio-Amplifier-Board/152990339062?_trkparms=aid%3D222007%26algo%3DSIM.MBE%26ao%3D2%26asc%3D52945%26meid%3D8755282de2e344a7a138da4aa3003cc8%26pid%3D100005%26rk%3D5%26rkt%3D12%26mehot%3Dpp%26sd%3D142809386353%26itm%3D152990339062&_trksid=p2047675.c100005.m1851)

Annoyingly they all seem to be sourced from China, despite a few claiming to be supplying from the UK (unless they've got UK warehousing, but that seems unlikely), so it'll take a while for us to get hold of any to test with.

**Comment by amcewen** [comment]
I've built up a circuit with the MAX9768, which works, but is disappointingly quiet.

**Comment by infovore** [comment]
@amcewen I think the MAX9768 has the same issues I had with the PAM8320: both chips have a supply range is listed as 4.5V->14V. The PAM8320 just about works at 5V but clips really easily and isn't very loud: it has no volume and _no headroom_. For an experiment, I ran a PAM8320 at 9V and it sounded brilliant, and loud. But you definitely don't get full performance at that very very low end of the voltage range.

**Comment by infovore** [comment]
I've also been looking into amplifier alternatives: firstly, a few chips from DiodesZetex:

- the [PAM8403](https://www.diodes.com/assets/Datasheets/PAM8403.pdf) and [PAM8406](https://www.mouser.com/ds/2/115/PAM8406-247303.pdf) are very similar. Two channel (left/right), Class D amp; no volume control internally (ie, you attenuate the input). 5V supply is in their sweet spot, and they describe themselves as "3W" amps into 4ohm load - which means, really, that they're about 1-and-a-bit-W tops into an 8ohm speaker. They are OK, but not quite as loud as the blue Not A Yamaha amp.
- the [PAM8320](https://www.mouser.co.uk/datasheet/2/115/PAM8320-273748.pdf) is similar to the Maxim chip Adrian tried: two-channel Class D going up to 10W. It's the latest generation of PAM, but I failed to read the datasheet in detail and at 5V input it clips easily and isn't very loud. At 9-12V they are great, but that's out of scope: I think keeping the unit operating off a 5V usb supply is sensible.

I'd also note that these require a decent number of external parts - filtering caps on PSU, RC filters on output, various caps bridging pins. Not huge, but I'm keeping an eye on them.

**Comment by infovore** [comment]
Next up: I'm going to investigate a TI [LM4860](http://www.ti.com/lit/ds/symlink/lm4860.pdf) "Boomer" - another chip designed for 5V supply, 1-and-a-bit-W output into 8ohm loads. Unlike the Diodes chips, it looks like I can increase the gain on the TI if I want to up to 10x, which is useful from a volume perspective - we still attenuate on input, but have the possibility to set the maximum output gain higher. 

It's a class-AB amp. One thing that is useful about it is as well as a 'shutdown' pin (for massive shutdown when running on battery) it also has a 'headphone disconnect' pin, meaning that if you have a stereo headphone jack with normalising pins, you can easily disable the amp when headphones are plugged in - I forget if the Brain has a headphone jack, and if it's just wired to the Pi in parallel. Anyhow, this way around, you could potentially have a volume knob for both amp/headphones, and just swap ins/outs on connection.

I should know more about this chip by Friday 27th. Again, this is SOIC16, which means that it's going to be hand-solderable in small numbers. (I find TSSOP/QFP to be microscope or magnifier jobs, which slows them down a lot, but SOIC16/0805 can be banged out by hand quite easily for smaller quantities)

**Comment by infovore** [comment]
And finally - regarding the amp chip, at the moment, I'm exploring basically building a replacement stand-alone board - keeping the amplifier/volume component a separate unit, with screwterms for all key connections. We've still got the capacity to make something that will fit precisely, but means we can move to a single "monolithic" brain PCB in a second step.

**Comment by amcewen** [comment]
@infovore I'm not sure that's the case with the MAX9768 - I didn't do lots of testing with it at higher voltages, but did try it ~12-14V range and didn't get much more from it.

There is a headphone jack on the box, which is wired in between the Pi and the amp and breaks the connection to the amp if you plug something into it.  The volume control only affects the amp (at present).

And stand-alone board is fine.  This issue was raised as a longer-term idea to explore, when we thought we had the amp situation resolved for the short term...

**Comment by infovore** [comment]
OK, we can just leave the headphone connection as is for now. Just flagging we can have 'actual' amp muting automatically off it.

**Comment by infovore** [comment]
Maybe got a winner here:

the TI [LM4860](http://www.ti.com/lit/ds/symlink/lm4860.pdf) turned up and I built it up. Worked first time. Nice and loud off 5V; all the extra components are lightweight passives - it's a differential output so filtering beads, etc, aren't really needed between it and the speaker. I set the gain to about 20 and it's fine, just clipping a tad at the top end when you drive it hard.

Also, the board comes out small - 42.55mm wide by 25mm high:

![image](https://user-images.githubusercontent.com/1103/43262427-fd630e20-90d7-11e8-8cad-0b81046aa054.png)

(that's literally what I had on my bench - those screwterms are 3.5mm spaced just because it's what I have lying around).

Schematic:

![image](https://user-images.githubusercontent.com/1103/43262543-689efc80-90d8-11e8-88b6-427e58474515.png)

**Comment by amcewen** [comment]
Excellent news!  Good to know we've got a candidate for a replacement.  And easier soldering and not-too-many passives are good features too :-)

**Comment by infovore** [comment]
@amcewen I might just get a bunch of those ordered today? They'd arrive by Monday, it looks like.

**Comment by amcewen** [comment]
@infovore how loud is loud for them?  Was chatting to @george08 just earlier and wondering if it's worth her doing a quick volume test first?

**Comment by amcewen** [comment]
Actually, two more questions:
  1. Are you assuming a panel mounted pot for that PCB, and if so does it mount directly to the PCB?  That'd be preferable for us, but I can't tell from the picture if the mounting holes for it are near enough to the edge of the PCB for that
 1. Is that from Euro Circuits?  (Curious, rather than concerned/opinionated)

**Comment by infovore** [comment]
@amcewen -

> Are you assuming a panel mounted pot for that PCB, and if so does it mount directly to the PCB? That'd be preferable for us, but I can't tell from the picture if the mounting holes for it are near enough to the edge of the PCB for that

Yes: on the 'right' of the above image, the two large holes and the three in a row beneath it are a panel-mounted pot on the _other side of the board. So: screwterms and components on the back side, pot coming out the other side, with a screw mount to the front of the brain. Does that make sense? I took measurements and think this will fit in place with no bending/bodging around the Pi.

I added a hole for another screw on the offchance it was necessary, but wasn't expecting it to be so. Would just give you some stability.

> Is that from Euro Circuits?

the screengrab is just from EAGLE's "manufacturing" tab; if I got some fabbed, I'd do it at AllPCB who are doing five-day turnarounds, and five-six of these boards would be $10. (They are clearly doing this as a loss-leader for custom).

**Comment by amcewen** [comment]
Ah, so the shaft of the pot is perpendicular to the PCB rather than parallel to it?  I'm happy to trust that you've got it in hand if it's something you've thought of, was mostly checking it had been a consideration.  

That does seem a good price for that turnaround!  All sounds good, get them ordered :ok_hand:

**Comment by infovore** [comment]
@amcewen yep, looks like this:

<img width="557" alt="image" src="https://user-images.githubusercontent.com/1103/43577908-2b06fc7a-9645-11e8-937b-5c503f77cc6f.png">

<img width="418" alt="image" src="https://user-images.githubusercontent.com/1103/43577919-379b3910-9645-11e8-977f-5f6fab0b085c.png">

**Comment by amcewen** [comment]
Nice!  I do like how PCB software has got 3D rendering these days (modulo having models for all your parts... :-)

**Comment by infovore** [comment]
OK - having demoed this to George, I'm going to order a prototype run of 5 boards, and parts for them.

I've also got a [BOM at Octopart](https://octopart.com/bom-tool/EVtU92Xq) which includes absolutely everything except the pot. At volume of 100, with a ~£1 pot and a board that AllPCB quote at 43 cents for 100, BOM comes in at £4-£5.

(As for renders - well, that's really just CAD software that happens to integrate with PCB software, but it sure does help visualise stuff spatially!)

**Comment by amcewen** [comment]
The new amp is nice!  Have you got details of the volume pot?  That's not listed on the BOM at present.  

And can I get the source files for the PCB - both for future reference, and also to tweak it to use 5.08mm pitch screw terminals (a [6-way](https://uk.farnell.com/multicomp/mc000038/terminal-block-wire-to-brd-6pos/dp/2008008?st=screw%20terminal) rather than 3 2-way to reduce the number of components to put on the board, but mostly a slight size increase because of the pre-crimped dupont connector wires we'll be inserting into them)

**Comment by george08** [comment]
@infovore can we please have a file for the PCB? it's time to place an order for 80 of 'em. Also curious about your Knob Man...

**Comment by infovore** [comment]
Hello! I'm on vacation in a farmhouse in Italy with poor connectivity at the minute, hence quiet. Back on Monday - can I get you some files then? I also have to patch up the logo. If you absolutely need it before Monday, I can see what I can do, but internet here is a bit crap.

Pots: the pot is a [9mm Alpha pot](https://www.thonk.co.uk/shop/alpha-9mm-pots/). That link is where I get them from - Thonk, in Brighton. £1.02 ex vat for quantities of 50+. Really high quality, metal shaft.

**Comment by infovore** [comment]
Back!

@amcewen : you can have EAGLE sch/brd files asap, along with gerbers, if you'd like. I can give you the 1.0 files - what I used to make the version I have - instantly. If you want a version with my first-pass for a .2in-spaced header, I can happily wrap that up asap, and then they're yours - but no worries if you just want the 1.0 files today. Regarding the volume pots I used - link above - I just used 10k audio taper (log), I _think_; resistance doesn't hugely matter, it's just being a voltage divider.

Around all day with clients but can answer things here.

**Comment by amcewen** [comment]
Welcome back @infovore!  We had a planning session while I was in London at the end of last week, and we're holding off ordering the PCBs until next week, after I've tested a couple of other things, so there's no immediate rush for it.  Thanks!

**Comment by george08** [comment]
@infovore can you please specify exactly which potentiometer I'm buying? (there are about 20 listed on the page)...?

**Comment by infovore** [comment]
@george08 of course, sorry I didn't clarify above:

On [this page](https://www.thonk.co.uk/shop/alpha-9mm-pots/), `VERTICAL: A10K AUDIO` is what you want to purchase. The bulk discount kicks in automatically when you order a quantity.

**Comment by george08** [comment]
Also, @infovore - who's the volume knob dude you like, please?

**Comment by amcewen** [comment]
And while you're here @infovore, we're getting to the ordering PCB stage, so if you can share the Eagle files (either for your original, or the 0.2"-spacing revision if that's in a reasonable state) that would be ace.  (either email them over, or stick them in this repo in [Physical](https://github.com/museuminabox/brain/tree/master/Physical)/amp (or similar), if you've got access.  Whichever's easiest.

**Comment by infovore** [comment]
@amcewen - you've got an email with the files, ping me if it's no good.

@george08 - I get my knobs from bitsbox - [this is the knob I tend to use](https://www.bitsbox.co.uk/index.php?main_page=product_info&cPath=94_108_111&products_id=849) but he has [many others](https://www.bitsbox.co.uk/index.php?main_page=index&cPath=94_108)

**Comment by amcewen** [comment]
Doing some digging into possible I2S DACs to use so we can eliminate the wiring between the Pi and amp, the PCM5102 and PCM5122 both look like good candidates (not many components needed, evidence of it already working on a Pi, ~£2/chip).  I've ordered a couple of PCM5102 breakout boards to try, and there's a [PCM5122 breakout board](https://oshpark.com/shared_projects/MmVdE7fl) available too, but we'd have to solder the parts on ourselves.

**Comment by amcewen** [comment]
All the work is done for this on the v1 box design.
